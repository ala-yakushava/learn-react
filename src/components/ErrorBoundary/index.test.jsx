import React from 'react';
import renderer from 'react-test-renderer';

import ErrorBoundary from './index';

describe('Components - ErrorBoundary', () => {
  const Child = () => new Error('Something went broke!');

  test('should match snapshot without error', () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <div>Content</div>
        </ErrorBoundary>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot with error', () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <Child />
        </ErrorBoundary>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
