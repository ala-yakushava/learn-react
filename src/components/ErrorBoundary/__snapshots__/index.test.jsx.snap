// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - ErrorBoundary should match snapshot with error 1`] = `
<div
  className="ErrorBoundary"
>
  <h2>
    Something went wrong.
  </h2>
  <details
    style={
      Object {
        "whiteSpace": "pre-wrap",
      }
    }
  >
    Error: Objects are not valid as a React child (found: Error: Something went broke!). If you meant to render a collection of children, use an array instead.
    in Child
    in ErrorBoundary
    <br />
    
    in Child
    in ErrorBoundary
  </details>
</div>
`;

exports[`Components - ErrorBoundary should match snapshot without error 1`] = `
<div>
  Content
</div>
`;
